flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManager.java

```diff
@@ -104,12 +192,40 @@ public void shutdown() {
 
             registeredPartitions.clear();
 
+            requestPartitionNotifiers.clear();
```

Comment:
Should we send `PartitionNotFoundException` to downstream?

Reply:
I think that's a good point

Reply:
I think we don't need to send any message to downstream for two reason:
1. When `ResultPartitionManager.shutdown` is called, it means the TM will exit and the job will fail, the job will enter the failover process and restart the downstream tasks
2. In the original processing, the downstream may not receive `PartitionNotFoundException` from upstream when `ResultPartitionManager` is shutdown